packages/dev/core/src/Misc/snapshotRenderingHelper.ts

```diff
@@ -0,0 +1,228 @@
+import type { Scene } from "core/scene";
+import type { AbstractMesh } from "core/Meshes/abstractMesh";
+import type { Mesh } from "core/Meshes/mesh";
+import type { Nullable } from "core/types";
+import type { Observer } from "./observable";
+import type { AbstractEngine } from "core/Engines/abstractEngine";
+import { Constants } from "core/Engines/constants";
+import type { WebGPUPipelineContext } from "core/Engines/WebGPU/webgpuPipelineContext";
+import { BindMorphTargetParameters } from "core/Materials/materialHelper.functions";
+import type { WebGPUDrawContext } from "core/Engines/WebGPU/webgpuDrawContext";
+import { WebGPUShaderProcessor } from "core/Engines/WebGPU/webgpuShaderProcessor";
+import type { EffectLayer } from "core/Layers/effectLayer";

```

Comment:
I think it is much easier to read and understand if we separate the type only imports from the runtime imports, and consolidate the type only imports to import from index (since it has no bundler or runtime impact).

Also I think ideally we don't want to pull in any WebGPU specific stuff that has a bundler or runtime impact, so in the block below I also changed that. See my other comment further down.

```suggestion
import type { AbstractEngine, AbstractMesh, EffectLayer, Mesh, Nullable, Observable, Scene, WebGPUDrawContext, WebGPUPipelineContext, WebGPUShaderProcessor } from "core/index";

import { Constants } from "core/Engines/constants";
import { BindMorphTargetParameters } from "core/Materials/materialHelper.functions";

```